Fix Race Condition in simbody-visualizer

mmatthe opened the pull request:
In rare circumstances (it happened to me on one of two machines), the simbody-visualizer would not run with OpenSim examples (I was testing against the [checkEnvironment](https://github.com/opensim-org/opensim-core/tree/main/OpenSim/Examples/checkEnvironment) example). The sympton was that simbody-visualizer would shortly show up, but crash immediately. 

I figured out this is due to a race condition in `simbody-visualizer.cpp`:

It can happen that during the rendering of the menus or the sliders a new command is for adding another menu is received. In this case, the `menus` vector can become reallocated, making the current object invalid.

This commit puts the scene lock around rendering the menus and sliders, to ensure that a menu is either added or drawn.

The bug also exists in the official apt-packages for libsimbody3.7.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/simbody/simbody/838)
<!-- Reviewable:end -->

sherm1 reviewed (approved):
Awesome! Thanks, Max. A couple of formatting nits, PTAL.
Otherwise, <img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

> @sherm1 reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/simbody/simbody/838#-:-OYCGhDY3Dx9plguiwuV:bpzlica)* status: all files reviewed, 2 unresolved discussions (waiting on @mmatthe)

___
*[`Simbody/Visualizer/simbody-visualizer/simbody-visualizer.cpp` line 1573 at r1](https://reviewable.io/reviews/simbody/simbody/838#-OYCGRNQE5bvOPEsf92M:-OYCGRNQE5bvOPEsf92N:b-n515wv) ([raw file](https://github.com/simbody/simbody/blob/a7bd0a49003c3e36c25ca3d500d825e16638e233/Simbody/Visualizer/simbody-visualizer/simbody-visualizer.cpp#L1573)):*
> ```cpp
> 
>     {
>         std::unique_lock<std::mutex> lock(sceneMutex); //------- LOCK SCENE -------
> ```

nit: keep line length to 80 per Simbody style guide
___
*[`Simbody/Visualizer/simbody-visualizer/simbody-visualizer.cpp` line 1598 at r1](https://reviewable.io/reviews/simbody/simbody/838#-OYCGJ6-EsWTm0JMGlKy:-OYCGJ6-EsWTm0JMGlKz:bji6hx6) ([raw file](https://github.com/simbody/simbody/blob/a7bd0a49003c3e36c25ca3d500d825e16638e233/Simbody/Visualizer/simbody-visualizer/simbody-visualizer.cpp#L1598)):*
> ```cpp
>         for (int i = 0; i < (int) sliders.size(); i++)
>             slidery = sliders[i].draw(slidery);
>     } /// ------------- UNLOCK SCENE ------------------
> ```

nit: should be `//` rather than `///` (Doxygen comment)


<!-- Sent from Reviewable.io -->

mmatthe commented:
thanks for the review! I adjusted the comment style accordingly. On a sidenote, the crash also just appeared under Ubuntu 22.04 and was also fixed after pulling this branch.

sherm1 reviewed (approved):
> @sherm1 reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/simbody/simbody/838)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mmatthe)



<!-- Sent from Reviewable.io -->
